File: GTP3/gtp1.cpp
#include<iostream>
#include<string.h>

using namespace std;

int main() {
        struct Equipos {
                char nombre[20];
                int goles_realizados;
                int goles_recibidos;
        };
        
        Equipos eq[8], aux[8];
        int CANTIDAD = 0;

        for (int i = 0; i < 8; i++) {
                cout << "Ingrese el nombre del equipo: ";
                cin.ignore();
                cin.getline(eq[i].nombre, 20, '\n');

                cout << "Ingrese goles hechos: ";
                cin >> eq[i].goles_realizados;
                
                cout << "Ingrese goles recibidos: ";
                cin >> eq[i].goles_recibidos;

                if(eq[i].goles_realizados > eq[i].goles_recibidos) {
                        aux[CANTIDAD] = eq[i];
                        CANTIDAD++;
                }
        }

        for (int i = 0; i < CANTIDAD; i++) {
                cout << "------ DATOS EQUIPO "<< i+1 <<" ------" << endl << endl;
                cout << "Nombre: " << aux[i].nombre << endl;
                cout << "Goles realizados: " << aux[i].goles_realizados << endl;
                cout << "Goles recibidos: " << aux[i].goles_recibidos << endl;
        }

        return 0;
}

File: GTP3/gtp1.py
import random as r

EMPRESAS:str = ["Eare", "Centidel", "Browsedrive", "Agimba", "Browsetype", "Buzzdog", "Eazzy", "Babbleopia", "Brainsphere", "Feedmix", "Brightdog", "Abatz", "Eamia", "Buzzster", "Avamba", "Feedfire", "Aivee", "Feedspan", "Feedfish", "Devify", "Browsebug", "Blogtag", "Demivee", "Brainlounge", "Feednation", "Eimbee", "Divanoodle", "Bluezoom", "Buzzbean", "Dablist", "Brightbean", "Dynabox", "Fivespan", "Bluejam", "Fanoodle", "Blognation", "Browsezoom", "Bubbletube", "Cogidoo", "Dabtype", "BlogXS", "Cogilith", "Camimbo", "Dynazzy", "Divavu", "Centizu", "Aimbo", "Fatz", "Dynava", "Centimia", "Einti", "Edgeblab", "Demizz", "Buzzshare", "Eidel", "Ainyx", "Babbleblab", "Dabjam", "Eadel", "Blogtags", "Devpoint", "Devshare", "Ailane", "Browsecat", "Edgeclub", "Cogibox", "Eabox", "Fivebridge", "Chatterbridge", "Avavee", "Camido", "Aibox", "Avamm", "Brainverse", "Edgepulse", "Babbleset", "Devbug", "Dabfeed", "Dabvine", "Aimbu", "DabZ", "Flashdog", "Abata", "Browseblab", "Flashpoint", "Edgetag", "Eire", "Avaveo", "Fivechat", "Brainbox", "Divape", "Eayo", "Edgeify", "Fiveclub", "Feedbug", "Devpulse", "Chatterpoint", "Digitube", "Devcast", "Blogspan"]

class Equipo:
        nombre:str
        goles_realizados:int
        goles_recibidos:int

        def __init__(self, nombre:str, greal:int, grec:int):
                self.nombre = nombre
                self.goles_realizados = greal
                self.goles_recibidos = grec
                
        def __repr__(self) -> str:
                return f"Nombre: {self.nombre}\nGoles realizados: {self.goles_realizados}\nGoles recibidos: {self.goles_recibidos}"

def main():
        eq:list[Equipo] = [Equipo(r.choice(EMPRESAS), r.randint(0,6), r.randint(0,6)) for n in range(8)]
        eq_:list[Equipo] = [e for e in eq if e.goles_realizados > e.goles_recibidos]

        for i in range(len(eq_)):
                print(eq_[i])
                print()

if __name__ == '__main__':
        main()
